util/src/zip_eq.rs

```diff
@@ -0,0 +1,63 @@
+/// An iterator which iterates two other iterators of the same length simultaneously.
+///
+/// Equality of the lengths of `a` abd `b` are at construction time.
```

Comment:
Whoops, I clearly wrote this comment too quickly.
```suggestion
/// Equality of the lengths of `a` and `b` are checked at construction time.
```